lib/ansible/modules/source_control/git.py

```diff
@@ -250,6 +256,19 @@
 from ansible.module_utils._text import to_native
 
 
+def relocate_repo(module, repo_dir, old_repo_dir, worktree_dir=None):
+    if os.path.exists(repo_dir):
+        module.fail_json(msg='Separate-git-dir path %s already exists.' % repo_dir)
+    if worktree_dir:
+        dot_git_file_path = os.path.join(worktree_dir, '.git')
+        try:
+            shutil.move(old_repo_dir, repo_dir)
+            with open(dot_git_file_path, 'w') as dot_git_file:
+                dot_git_file.write('gitdir: %s' % repo_dir)
+        except (IOError, OSError) as err:
```

Comment:
what will happen if modification of `.git` file fails while `.git/` dir is already moved? I think we could try to roll it back.

Reply:
yes, I will figure out how to roll back.